cli/mender-artifact/write.go

```diff
@@ -517,10 +528,26 @@ func getDeviceSnapshot(c *cli.Context) (string, error) {
 		args = append(args, "-p", port)
 	}
 	args = append(args, userAtHost)
+	useSudo, _ := hasCommand(append(args, "sudo", "--help"))
+
+	_, out := hasCommand(append(args, "id", "-u"))
```

Comment:
A problem with this approach is that you have to type your ssh password three times. Can you work this into one call instead? For example something like:

```
args = append(args, "/bin/sh", "-c",
		`'if [ $(id -u) = 0 ]; then SUDO_CMD=sudo; else SUDO_CMD=; fi; $SUDO_CMD echo "`+sshInitMagic+`" && $SUDO_CMD mender snapshot dump | cat'`)
```

Untested and *probably* doesn't work exactly as I put it, but you get what I mean!

I think it is not important to handle the case where `id != 0` and `sudo` doesn't exist, because in this case the operation is guaranteed not to work anyway.

Reply:
you definitely broke all the fun here. I will try though.

Reply:
Hehe, I know. But the User Is Always Right™! :slightly_smiling_face:

Reply:
hehe. right! I was _that_ close to actually ask for the password and play around with it, but in the end I followed [your advice](https://github.com/mendersoftware/mender-artifact/pull/305).